WebDriverAgentLib/Commands/FBElementCommands.m

```diff
@@ -555,10 +555,11 @@ + (NSArray *)routes
 
 + (id<FBResponsePayload>)handleKeys:(FBRouteRequest *)request
 {
+  XCUIApplication *app = request.session.activeApplication ?: FBApplication.fb_activeApplication;
   NSString *textToType = [request.arguments[@"value"] componentsJoinedByString:@""];
   NSUInteger frequency = [request.arguments[@"frequency"] unsignedIntegerValue] ?: [FBConfiguration maxTypingFrequency];
   NSError *error;
-  if (![FBKeyboard typeText:textToType frequency:frequency error:&error]) {
+  if (![app fb_typeText:textToType shouldClear:NO frequency:frequency error:&error]) {
```

Comment:
Wouldn’t calling `activeApplication` and perform snapshot-based operations as part of the flow cause unneeded performance penalty? Previously the `handleKeys` command was element agnostic.

Reply:
I also had some doubts there. Let me add a flag to skip the focus check, so we would only rely on the current UI state without doing any additional checks